pkg/provider/configuration.go

```diff
@@ -383,7 +383,8 @@ func BuildTCPRouterConfiguration(ctx context.Context, configuration *dynamic.TCP
 			if len(configuration.Services) > 1 {
 				delete(configuration.Routers, routerName)
 				loggerRouter.
-					Error("Could not define the service name for the router: too many services")
+					Errorf("Router %s cannot be linked automatically with multiples Services : %q",
+						routerName, reflect.ValueOf(configuration.Services).MapKeys())
```

Comment:
@mloiseleur While using `reflect` here looks pretty handy, it is not common for us to use reflection in such simple cases. To be consistent with the codebase, I suggest creating the service names slice instead.

Reply:
I'm unsure what you expect. I tried something with 676c8d723. Is this what you have in mind ?

Reply:
Sure, it is what I had in mind 👍